### PR TITLE
IMU: parameterize IMU integration time (IMU_INTEG_RATE)

### PR DESCRIPTION
**Pull request by dagar**
This effectively controls the `vehicle_attitude` publication rate and subsequently how fast the attitude controllers run. It's not something I expect regular users to to change, but this is better than having a magic hard coded number.

We might want to consider lowering this by default for FW.

### CODE REVIEW

**Comment by dagar** [comment]
Historically the IMU integration time was  4000 us (250 Hz) and only very recently changed to 2500 us (400 Hz). We could also consider if 5000 us (200 Hz) is a better default.

**Comment by dagar** [comment]
Changing the integration from 2500 us -> 5000 us saves about 7% cpu on F4 boards.

**Comment by dagar** [comment]
Yet another factor to consider for the default is the impact on default logging rate and ekf2 replay.

On a typical simple multicopter configuration the nominal logging rate changes from ~ 47 KiB/s (IMU integration 2500 us) down to ~32 KiB/s (IMU integration 5000 us).

File: src/lib/drivers/gyroscope/PX4Gyroscope.cpp [hunk 1]
```diff
@@ -122,11 +125,19 @@ void PX4Gyroscope::set_device_type(uint8_t devtype)
...
-	const uint32_t update_interval = 1000000 / rate;
+	_update_rate = math::constrain((int)rate, 50, 32000);
+	const uint32_t update_interval = 1000000 / _update_rate;
+
+	// constrain IMU integration time 1-20 milliseconds (50-1000 Hz)
+	int32_t imu_integration_time_us = math::constrain(_param_imu_integ_time.get(), 1000, 20000);
+
+	if (imu_integration_time_us != _param_imu_integ_time.get()) {
+		_param_imu_integ_time.set(imu_integration_time_us);
+		_param_imu_integ_time.commit_no_notification();
```
Comment (MaEtUgR) [inline src/lib/drivers/gyroscope/PX4Gyroscope.cpp:136]: Isn't it a problem that all Gyroscope and Accelerometer drivers would try to correct the out of range parameter more or less at the same time?

Comment (MaEtUgR) [reply src/lib/drivers/gyroscope/PX4Gyroscope.cpp:136]: I would at least define the [1000, 20000] interval shared to make sure it's always in sync.

Comment (dagar) [reply src/lib/drivers/gyroscope/PX4Gyroscope.cpp:136]: Yes I didn't like that part either. Very soon (early in the v1.12 release cycle) the plan is to gut all of this code from PX4Accelerometer/PX4Gyroscope and move it "downstream" to sensors/vehicle_imu where we can effectively synchronize accel + gyro.

**Comment by MaEtUgR** [comment]
One more question I forgot to ask in the review: Is there a specific reason why the new parameter is an interval while all others I know of are rates in Hz? Just thinking about the user perspective.

**Comment by dagar** [comment]
> One more question I forgot to ask in the review: Is there a specific reason why the new parameter is an interval while all others I know of are rates in Hz? Just thinking about the user perspective.

Not really, I was thinking about it from the perspective of what was already hard coded and aligning with the current ecl/EKF filter update expressed as an interval (10 ms). We reset the integrator and publish the integrated value every ____ microseconds.

**Comment by dagar** [comment]
Default changed to 5000 us.

**Comment by dagar** [comment]
SITL appears to be broken with 5000 us IMU integration.

**Comment by julianoes** [comment]
@dagar we might have to adapt the gazebo side too:

**Comment by dagar** [comment]
Testing with sitl_gazebo and jmavsim updated as well.
 - https://github.com/PX4/sitl_gazebo/pull/474
 - https://github.com/PX4/jMAVSim/pull/118

**Comment by dagar** [comment]
This was tested by @mhkabir in https://github.com/PX4/Firmware/issues/14735.

File: src/modules/sensors/vehicle_imu/imu_parameters.c [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+* IMU integration period.
```
Comment (dagar) [inline src/modules/sensors/vehicle_imu/imu_parameters.c:35]:
```suggestion
* IMU integration time.
```

**Comment by dagar** [comment]
CI SITL MC_avoidance and MC_safe_landing still failing due to the sitl_gazebo version.

**Comment by dagar** [comment]
I've made 2 minor changes to the PR.
1. renamed IMU_INTEG_TIME -> IMU_INTEG_RATE (I think it's slightly easier for most people to think in terms of rate)
2. Dropped the sitl_gazebo and jmavsim changes for now until we can resolve the performance problems https://github.com/PX4/sitl_gazebo/pull/474.

**Pull request merged**